API-source/models/teamgantt.php

```diff
@@ -19,43 +21,115 @@ public function fetch_teamgantt_todolist_id($logged_in_user) {
       }
     }
 
-    public static function fetch_list($logged_in_user, $query) {
-      if($query == 'today') {
-        $teamgantt_tasks = API::run($logged_in_user, 'GET', 'https://api.teamgantt.com/v1/tasks?today');
+    public static function synced_tasks($logged_in_user) {
+      $todos = array();
+      $result = mysql_query("SELECT
+                              t.id,
+                              t.todolist_id,
+                              t.name,
+                              t.percent_complete,
+                              t.is_big_rock,
+                              t.is_current,
+                              t.created_at,
+                              t.updated_at,
+                              t.teamgantt_id,
+                              t.teamgantt_meta
+                                FROM todolists AS tl
+                                JOIN todos AS t ON t.todolist_id = tl.id
+                                WHERE tl.user_id = '".$logged_in_user->id."' AND t.teamgantt_id != ''
+                                ORDER BY t.created_at ASC");
+      while($row = mysql_fetch_assoc($result)) {
+        array_push($todos, new Todo(array('id' => (int) $row['id'],
+                                          'todolist_id' => (int) $row['todolist_id'],
+                                          'name' => $row['name'],
+                                          'percent_complete' => (int) $row['percent_complete'],
+                                          'is_big_rock' => (bool) $row['is_big_rock'],
+                                          'is_current' => (bool) $row['is_current'],
+                                          'created_at' => $row['created_at'],
+                                          'updated_at' => $row['updated_at'],
+                                          'teamgantt_id' => $row['teamgantt_id'],
+                                          'teamgantt_meta' => $row['teamgantt_meta']
+                                      )));
+      }
+      return $todos;
+    }
+
+    public static function sync($logged_in_user) {
```

Comment:
New function to sync. This will load all tasks that have been synced with TG, then put them into an array. Then, we create another array 'teamgantt_ids' that is used to map teamgantt task id to the array key for that todo.

We then loop through the results from TG. If the task id does not exist as a key in 'teamgantt_ids' we create a new todo and save it. If it does, we check to see if the name or percent complete were updated. If they were, we update them, then save.